_posts/2024-05-28-edb-and-kasten.markdown

```diff
@@ -0,0 +1,295 @@
+
+```
+kubectl apply -f https://get.enterprisedb.io/cnp/postgresql-operator-1.20.2.yaml
+```
+
+This will create the operator namespace where the controller will be running.
+
+## Create an EDB cluster, a client and some data 
+
+
+```
+kubectl create ns edb
+kubectl apply -f https://github.com/michaelcourcy/edb-kasten/raw/main/cluster-example-2.yaml -n edb
+```
+
+Wait for the cluster to be fully ready.
+```
+kubectl get clusters.postgresql.k8s.enterprisedb.io -n edb
+NAME              AGE   INSTANCES   READY   STATUS                     PRIMARY
+cluster-example   19m   3           3       Cluster in healthy state   cluster-example-1
+```
+
+
+Install the cnp plugin if you haven't it yet 
+```
+curl -sSfL \
+  https://github.com/EnterpriseDB/kubectl-cnp/raw/main/install.sh | \
+  sudo sh -s -- -b /usr/local/bin
+```
+
+Create a client certificate to the database
+```
+kubectl cnp certificate cluster-app \
+  --cnp-cluster cluster-example \
+  --cnp-user app \
+  -n edb 
+```
+
+Now you can create the client 
+```
+kubectl create -f https://github.com/michaelcourcy/edb-kasten/raw/main/client.yaml -n edb 
+```
+
+Create some data 
+```
+kubectl exec -it deploy/cert-test -- bash
+psql " $DATABASE_URL "
+\c app
+DROP TABLE IF EXISTS links;
+CREATE TABLE links (
+	id SERIAL PRIMARY KEY,
+	url VARCHAR(255) NOT NULL,
+	name VARCHAR(255) NOT NULL,
+	description VARCHAR (255),
+        last_update DATE
+);
+INSERT INTO links (url, name, description, last_update) VALUES('https://kasten.io','Kasten','Backup on kubernetes',NOW());
+select * from links;
+\q
+exit
+```
+
+## Add the backup decorator annotations to the cluster 
+
+You can skip this part if you create the cluter from the previous section because with [cluster-example-2](https://github.com/michaelcourcy/edb-kasten/raw/main/cluster-example-2.yaml) the cluster-example already include the kasten addon.
+
+If you have not it yet just had this annotation in your cluster CR 
+```
+    "k8s.enterprisedb.io/addons": '["kasten"]'
+```
+
+If your version of EDB is old and does not support the kasten addons you can create all the annotations and labels manually 
+you have an example in [cluster-example.yaml ](https://github.com/michaelcourcy/edb-kasten/raw/main/cluster-example.yaml). 
+
+
+## Install the edb blueprint
+
+```
+kubectl create -f https://github.com/michaelcourcy/edb-kasten/raw/main/edb-hooks.yaml
+```
+
+## Create a backup policy with the exclude filters and the hooks 
+
+Create a Kasten policy for the edb namespace: set up a location profile for export and kanister actions. 
+
+### Add the exlude filters :
+
+```
+kasten-enterprisedb.io/excluded:true
+```
+
+![PVC exclude filters](/images/posts/2024-05-28-edb-and-kasten/exclude-filters.png)
+
+
+### Add the hooks :
+
+![Policy hooks](/images/posts/2024-05-28-edb-and-kasten/policy-hooks.png)
+
+
+## Launch a backup 
+
+Launch a backup, that will create 2 restorepoints a local and a remote.
```

Comment:
Launch a backup, that will create 2 restorepoints a local and a remote.  --> Launch a backup, this creates 2 restorepoints a local and a remote.